pkg/controller/postgresqldatabase/postgresqldatabase_controller.go

```diff
@@ -96,89 +90,38 @@ func (r *ReconcilePostgreSQLDatabase) Reconcile(request reconcile.Request) (reco
 	reqLogger = reqLogger.WithValues("database", database.Spec.Name)
 	reqLogger.Info("Reconciling PostgreSQLDatabase")
 
-	// Resolve the password, is the value in a configMap or Secret or just a plain value
+	host, err := kube.ResourceValue(r.client, database.Spec.Host, request.Namespace)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
 	password, err := kube.ResourceValue(r.client, database.Spec.Password, request.Namespace)
 	if err != nil {
 		return reconcile.Result{}, err
 	}
 
 	// Ensure the database is in sync with the object
-	err = r.EnsurePostgreSQLDatabase(reqLogger, database.Spec.Name, password)
+	err = r.EnsurePostgreSQLDatabase(reqLogger, host, database.Spec.Name, password)
 	if err != nil {
 		return reconcile.Result{}, err
 	}
 	return reconcile.Result{}, nil
 }
 
-func postgresqlConnection(connectionString string) (*sql.DB, error) {
-	log.Info("Connecting to database", "url", connectionString)
-	db, err := sql.Open("postgres", connectionString)
-	if err != nil {
-		return nil, err
+func (r *ReconcilePostgreSQLDatabase) EnsurePostgreSQLDatabase(log logr.Logger, host, name, password string) error {
+	credentials, ok := r.hostCredentials[host]
+	if !ok {
+		return fmt.Errorf("unknown credentials for host %s", host)
 	}
-	err = db.Ping()
+	db, err := postgres.Connect(log, fmt.Sprintf("postgresql://%s:%s@%s?sslmode=disable", credentials.Name, credentials.Password, host))
```

Comment:
Is there a reason you don't want sslmode enabled?

Reply:
Currently I don't think we handle this. I'll get back to it as I'd like to do it by default, but when running test postgres instances SSL is not an option.

Reply:
I thought it might be related to the tests. 
Would be nice to have this addition but not a NEED for a merge of this.